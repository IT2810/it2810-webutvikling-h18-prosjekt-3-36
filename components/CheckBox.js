import React from 'react';
import {
    StyleSheet, View, Text
  } from 'react-native';
import Colors from '../constants/Colors';
import { Icon } from 'expo';


export default class CheckBox extends React.Component {
  render() {
    return (
        <View style={styles.container}>
            <View style={styles.outerBar}>
                <View style={{position: "relative", top: -3, right: 0.3, borderRadius: 24,  height: (this.props.checked ? 125 : 0) +"%", width: (this.props.checked ? 120 : 0) +"%"}}>
                        <Icon.Ionicons
                        name={(Platform.OS === 'ios' ? "ios-" : "md-") + "checkmark-circle"} // categories[this.props.category].icon 
                        size={28}
                        style={styles.icon}
                        color={this.props.color }
                        />
                </View>
            </View>
        </View>

    );
  }
}
const styles = StyleSheet.create({
    container: {
        margin: 10,
        marginTop: 10
    },
    icon: {
    },
    outerBar: {
      backgroundColor: "white",
      borderRadius: 24,
      width: 24,
      height: 24,
      borderWidth: 1,
      borderColor: 'gray',
    }
  });
