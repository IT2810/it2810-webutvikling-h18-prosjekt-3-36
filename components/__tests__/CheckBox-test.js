import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import CheckBox from "../CheckBox";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("CheckBox Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<CheckBox title="test" onPress={() => { }} store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});