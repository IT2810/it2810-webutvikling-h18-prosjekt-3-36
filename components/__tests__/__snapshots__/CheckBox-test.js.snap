// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "margin": 10,
      "marginTop": 10,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 24,
        "borderWidth": 1,
        "height": 24,
        "width": 24,
      }
    }
  >
    <Component
      style={
        Object {
          "borderRadius": 24,
          "height": "0%",
          "position": "relative",
          "right": 0.3,
          "top": -3,
          "width": "0%",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        name="ios-checkmark-circle"
        size={28}
        style={Object {}}
      />
    </Component>
  </Component>
</Component>
`;
